crates/database/src/order_execution.rs

```diff
@@ -1,33 +1,103 @@
 use {
-    crate::{auction::AuctionId, OrderUid},
+    crate::{auction::AuctionId, Address, OrderUid},
     bigdecimal::BigDecimal,
-    sqlx::PgConnection,
+    sqlx::{PgConnection, QueryBuilder},
+    std::collections::HashMap,
 };
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct FeeAsset {
+    pub amount: BigDecimal,
+    pub token: Address,
+}
+
 pub async fn save(
     ex: &mut PgConnection,
     order: &OrderUid,
     auction: AuctionId,
     block_number: i64,
     executed_fee: &BigDecimal,
+    executed_protocol_fees: &[FeeAsset],
 ) -> Result<(), sqlx::Error> {
+    let (protocol_fee_tokens, protocol_fee_amounts) = executed_protocol_fees.iter().fold(
+        (Vec::new(), Vec::new()),
```

Comment:
we can initialize these vectors with the capacity

Reply:
true, adjusted